app/assets/javascripts/services/planService.js

```diff
@@ -178,25 +223,30 @@ planner.factory('planService', ['Restangular', '_', function(Restangular, _) {
 
   // used in callbacks
   var addOrRemoveIntended = function(course) {
-    if (course.intended) {
-      _addToIntended(course);
+    // if (course.intended) {
+    //   _addToIntended(course);
+    // } else {
+    //   _removeFromIntended(course);
+    // }
```

Comment:
I have removed the functions that add a course to the visuals before the updated plan is returned from the backend. If you remember, there was an effect of the course moving to intended/completed area, then the list reorganizing itself. I prefer it this way, but have left it as comments for discussion.